tracing-subscriber/src/filter/env/field.rs

```diff
@@ -36,14 +36,23 @@ pub(crate) struct MatchVisitor<'a> {
     inner: &'a SpanMatch,
 }
 
-#[derive(Debug, Clone, PartialOrd, Ord, Eq, PartialEq)]
+#[derive(Debug, Clone, PartialOrd, PartialEq)]
 pub(crate) enum ValueMatch {
     Bool(bool),
+    F64(f64),
     U64(u64),
     I64(i64),
     Pat(Box<MatchPattern>),
 }
 
+impl Eq for ValueMatch {}
```

Comment:
Similarly to `Ord`, the `Eq` implementation is only correct if we ensure there are no `NaN` values present in the `ValueMatch::F64` variant. See my comment on the `Ord` impl for details.

Reply:
:+1: